Rubberduck.Parsing/ComReflection/ComParameter.cs

```diff
@@ -87,48 +93,60 @@ private void GetParameterType(TYPEDESC desc, ITypeInfo info)
             var vt = (VarEnum)desc.vt;
             TYPEDESC tdesc;
 
-            switch (vt)
+            if (vt == VarEnum.VT_PTR)
             {
-                case VarEnum.VT_PTR:
-                    tdesc = (TYPEDESC)Marshal.PtrToStructure(desc.lpValue, typeof(TYPEDESC));
-                    GetParameterType(tdesc, info);
-                    IsByRef = true;
-                    break;
-                case VarEnum.VT_USERDEFINED:
-                    int href;
-                    unchecked
-                    {
-                        href = (int)(desc.lpValue.ToInt64() & 0xFFFFFFFF);
-                    }
-                    try
+                tdesc = Marshal.PtrToStructure<TYPEDESC>(desc.lpValue);
+                GetParameterType(tdesc, info);
+                IsByRef = true;
+            }
+            else if (vt == VarEnum.VT_USERDEFINED)
+            {
+                int href;
+                unchecked
+                {
+                    href = (int)(desc.lpValue.ToInt64() & 0xFFFFFFFF);
```

Comment:
I think a comment is needed here to explain what you are doing here.

Reply:
Done, but might not be able to sync it with this PR without pulling in a load of other crap.

`//The href is a long, but the size of lpValue depends on the platform, so truncate it after the lword.`